app/models/settings/base.rb

```diff
@@ -0,0 +1,166 @@
+# This was taken from the rails-settings-cached gem and adapted to work with
+# more than one settings model by making the request cache class-specific. We
+# also removed features we don't need like cache scopes and readonly fields.
+#
+# See: https://github.com/huacnlee/rails-settings-cached
+module Settings
+  # rubocop:disable Security/YAMLLoad
+  class Base < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
+    self.abstract_class = true
+
+    class SettingNotFound < RuntimeError; end
+
+    class ProcetedKeyError < ArgumentError; end
+
+    PROTECTED_KEYS = %w[var value].freeze
+    SEPARATOR_REGEXP = /[\n,;]+/.freeze
+
+    class << self
+      def inherited(subclass)
+        super
+
+        # This defines a custom request cache based on ActiveSupport::CurrentAttributes.
+        # See: https://api.rubyonrails.org/classes/ActiveSupport/CurrentAttributes.html
+        # NOTE: An anonymous class doesn't work here due to implementation details of
+        # CurrentAttributes.
+        request_cache = Class.new(ActiveSupport::CurrentAttributes) { attribute :settings }
+        subclass.const_set(:RequestCache, request_cache)
+      end
+
+      def clear_cache
+        self::RequestCache.reset
+        Rails.cache.delete(cache_key)
+      end
+
+      def field(key, default: nil, type: :string, separator: nil, validates: nil)
```

Comment:
In the original code they signature was `key, **opts` but since this is a public method I opted for making the kwargs explicit.